lld/COFF/DriverUtils.cpp

```diff
@@ -246,6 +246,21 @@ void LinkerDriver::parseAligncomm(StringRef s) {
       std::max(ctx.config.alignComm[std::string(name)], 1 << v);
 }
 
+void LinkerDriver::parseStub(StringRef path) {
+  std::unique_ptr<MemoryBuffer> stub =
+      CHECK(MemoryBuffer::getFile(path), "could not open " + path);
+  size_t bufferSize = stub->getBufferSize();
+  const char *bufferStart = stub->getBufferStart();
+  // MS link.exe compatibility:
+  // 1. stub must be greater or equal than 64 bytes
+  // 2. stub must be 8-byte aligned
```

Comment:
You mean the size must be a multiple of 8? Does link.exe really require that? I suppose the PE header needs to be 8-byte aligned, but the linker could easily add padding after the stub.

Reply:
> You mean the size must be a multiple of 8? Does link.exe really require that?

Yes, I already tested that and confirmed that LINK.exe does not allow unaligned stub input. :)

Reply:
I don't feel too strongly about it, but unless there's a technical reason to reject such stubs, I think we should accept also non-8-byte-aligned programs and add any padding ourselves.

https://learn.microsoft.com/en-us/cpp/build/reference/stub-ms-dos-stub-file-name?view=msvc-170 says "any valid MS-DOS application [.exe file] can be a stub program", for example.

Reply:
Well, I am pretty sure we should keep `MZ` dos signature check. On the other hand, there's nothing prevents us to accept unaligned stubs, if we don't have to replicate MS linker bug-by-bug. It's implemented on [94ac28f183d994e0ba5f61a9fcc788369be95970 ](https://github.com/llvm/llvm-project/pull/122561/commits/94ac28f183d994e0ba5f61a9fcc788369be95970) and I'd love to hear feedback from you :) .